### DISCUSSION TITLE
Why "test.client" resets container replaced dependencies after request?

### DISCUSSION THREAD

**Discussion by c0nst4nt** [symfony/symfony#49106] [Q&A] [answered]
Hello, community! I have a question related to testing. I do remember times, when I could set some mock in container like that:

```
// code inside test suite, which extends WebTestCase 
$client = static::createClient();
$myMock = $this->createMock(SomeInterface::class);
$client->getContainer()->set(SomeInterface::class, $myMock);
```

And then do several requests, relying on that mock. Then just perform some asserts and check result. 
```
$this->httpClient->request('GET', '/example');
$this->httpClient->request('GET', '/example2');
```

In my current Symfony version (5.4) I can't do that. For the second request, my mock inside container is just being reset. Eventually my test case fails because of that. And I need setup container service again before each next request. Did somebody had such situation? I haven't found any issues or discussion about that. Maybe there were some info about that also. Can somebody clarify why it is like that? And since which moment that was implemented?

**Comment by stof** [comment, marked as answer]
By default, the client will reboot the kernel between requests. A reboot of the kernel replaces the whole container with a new one (and in this new one, no replacement has been done).

**Reply by c0nst4nt** [reply]
Okay, I got it, thanks! 
So, as you said "by default" - is there any option to be changed, to disable such behavior? Probably at Symfony version 4, it was different.

**Reply by stof** [reply]
There is a `disableReboot` method (and its sibling `enableReboot`) in the KernelBrowser class (which is what the `createClient` returns)

**Reply by c0nst4nt** [reply]
Great, thanks!
